src/main/java/study/StringAddCalculator.java

```diff
@@ -0,0 +1,49 @@
+package study;
+
+import java.util.Arrays;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.IntStream;
+
+class StringAddCalculator {
+
+    private static final String SPLIT_REGEX_FOR_SUM = "[,:]";
+    private static final String DELIMITER_REGEX_FOR_SUM = "//(.)\n(.*)";
+
+    private static final String EXCEPTION_MSG_UTILITY_CLASS = "Utility class";
+    private static final String EXCEPTION_MSG_NEGATIVE_INPUT = "negative input is not allowed";
+
+    private StringAddCalculator() {
+        throw new IllegalStateException(EXCEPTION_MSG_UTILITY_CLASS);
+    }
+    public static int sum(String inputNumbers){
+        if(isBlankOrNull(inputNumbers)){
+            return 0;
+        }
+        return toIntStream(toStringArray(inputNumbers)).reduce(0, StringAddCalculator::adder);
+    }
+
+    private static int adder(int prev, int current) {
+        if(current < 0){
+            throw new IllegalArgumentException(EXCEPTION_MSG_NEGATIVE_INPUT);
+        }
+        return prev + current;
+    }
+
+    private static String[] toStringArray(String inputNumbers){
+        Matcher m = Pattern.compile(DELIMITER_REGEX_FOR_SUM).matcher(inputNumbers);
```

Comment:
사소하지만 현재 메서드 호출할때마다 Pattern.compile()을 호출하고 있네요 :)
Pattern.compile 내부 구현을 보면
```java
// Pattern
public static Pattern compile(String regex) {
    return new Pattern(regex, 0);
}
```

매번 메서드 호출시 성능이 비싼 Pattern 인스턴스를 생성하고 있습니다.
Pattern 인스턴스는 한번 쓰고 버려지기 때문에 성능적으로 좋지 않기에
자바 doc에서도 한번 컴파일 후 재사용 할것을 권장하고 있습니다!

```
If a pattern is to be used multiple times, compiling it once and reusing
it will be more efficient than invoking this method each time
```

불필요한 객체 생성을 피하도록 개선해주시면 좋을 것 같아요 😄